06_puppy/n0npax/store_test.go

```diff
@@ -0,0 +1,121 @@
+package main
+
+import (
+	"testing"
+
+	tassert "github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+type storerImpl int
+
+const (
+	syncStorer storerImpl = iota
+	memStorer  storerImpl = iota
+)
+
+var (
+	puppy0 = func() Puppy {
+		return Puppy{
+			Breed:  "Norwegian Forest cat",
+			Colour: "Grey",
+			Value:  "$300",
+		}
+	}
+	puppy1 = func() Puppy {
+		return Puppy{
+			Breed:  "Maine Coon",
+			Colour: "Brown",
+			Value:  "€400",
+		}
+	}
+)
+
+type storerSuite struct {
+	suite.Suite
+	store Storer
+	impl  storerImpl
+}
+
+func (s *storerSuite) SetupTest() {
+	switch s.impl {
+	case syncStorer:
```

Comment:
Awesome work!! Great use of the SetupTest function.

Can you please extract the inserting of a puppy so it is outside the switch statement, that way you won't have to repeat it for each implementation. Saves a couple of lines of code.

Reply:
fixed Thanks for that!